llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -407,6 +410,23 @@ Expected<GroupRef> GroupRef::create(MCCASBuilder &MB,
   return get(B->build());
 }
 
+template <typename RefTy>
+static Expected<SmallVector<RefTy, 0>> findRefs(MCCASReader &Reader,
+                                                ArrayRef<cas::ObjectRef> Refs) {
+  SmallVector<RefTy, 0> TopRefs;
+  for (auto ID : Refs) {
+    auto Node = Reader.getObjectProxy(ID);
+    if (!Node)
+      return Node.takeError();
+    if (auto TopRef = RefTy::Cast(*Node))
+      TopRefs.push_back(*TopRef);
+  }
+  if (TopRefs.size())
+    return TopRefs;
```

Comment:
Since the return type here is `Expected<SmallVector>` instead of just `SmallVector`, I think this is one of those instances where a `return std::move(TopRefs);` is correct.
However, I am not a (C++) lawyer 😆  Could you give this a try and see if Clang emits a warning?

Reply:
IIRC you use std::move when it returns an Expected, and we are returning an Error.

Reply:
Oh in this case the question is about moving *the vector* into the Expected.
In other words, are we calling the constructor `Expected(SmallVector&&)` or `Expected(const SmallVector&)`?

Reply:
but it works either way